File: backend/src/main/java/ResumeBuilder/backend/Models/DatabaseModels/GuestAccountModel.java
package ResumeBuilder.backend.Models.DatabaseModels;

import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "guest_account")
public class GuestAccountModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    public int id;

    @Column(unique = true, nullable = false)
    public String session;

    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "user_account_id", nullable = false)
    public UserAccountModel userAccountModel;
}

File: frontend/models/GuestAccountModel.js
import UserAccountModel from "./UserAccountModel";
import { v4 as uuid } from "uuid";

export default class GuestAccountModel {
  id = uuid();
  session = "";
  userAccountModel = new UserAccountModel();

  constructor(data) {
    if (data != null) {
      this.id = data.id;
      this.session = data.session;
      this.userAccountModel = data.userAccountModel;
    }
  }
}
